src/ptvsd/_vendored/pydevd/_pydevd_bundle/pydevd_comm.py

```diff
@@ -1130,7 +1130,10 @@ def internal_get_exception_details_json(dbg, request, thread_id, set_additional_
             name = 'exception: type unknown'
             description = 'exception: no description'
 
-        stack_str = ''.join(traceback.format_list(frames))
+        if max_frames == 0:
+            stack_str = ''.join(traceback.format_list(frames))
+        else:
+            stack_str = ''.join(traceback.format_list(frames[-max_frames:]))
```

Comment:
Doesn't look like you need an `if` here, since `frames[-0:]` will get the whole thing just the same.